cpp/src_prims/linalg/gemm.h

```diff
@@ -180,5 +180,101 @@ void gemm(const math_t *a, int n_rows_a, int n_cols_a, const math_t *b,
        beta, cublas_h, stream);
 }
 
+/**
+ * @brief A wrapper for CUBLS GEMM function designed for handling all possible 
+ * combinations of operand layouts.
+ * It computes the following equation: Z = alpha . X * Y + beta . Z
+ * @tparam T Data type of input/output matrices (float/double)
+ * @param handle cublas handle
+ * @param Z output matrix of size M rows x N columns
+ * @param X input matrix of size M rows x K columns
+ * @param Y input matrix of size K rows x N columns
+ * @param _M number of rows of X and Z
+ * @param _N number of rows of Y and columns of Z
+ * @param _K number of columns of X and rows of Y
+ * @param isZColMajor Storage layout of Z. true = col major, false = row major
+ * @param isXColMajor Storage layout of X. true = col major, false = row major
+ * @param isYColMajor Storage layout of Y. true = col major, false = row major
+ * @param alpha scalar
+ * @param beta scalar
+ * @{
+ */
+template <typename T>
+void gemm(cublasHandle_t handle, T *z, T *x, T *y, int _M, int _N, int _K,
+          bool isZColMajor, bool isXColMajor, int isYColMajor, T alpha = T(1.0),
+          T beta = T(0.0)) {
```

Comment:
Why is this function not accepting an explicit cuda stream as input?

Reply:
I was unsure about stream parameter. Do we take stream and cublas handle as parameter or do we extract stream from the handle. I have seen both approaches being taken in the prims. I am okay with any approach. I'll add explicit stream,

Reply:
that's correct. We explicitly pass stream parameter to all prims methods. Eg: our templated gemm wrapper [here](https://github.com/rapidsai/cuml/blob/branch-0.12/cpp/src_prims/linalg/cublas_wrappers.h#L159)